_posts/2025-02-13-rfprs.md

```diff
@@ -0,0 +1,372 @@
+---
+title:  Reviewer First Pull Requests
+classes: wide
+---
+
+Pull requests continue to be the primary mechanism by which we collaborate and contribute code.
+Despite this we often will put as little effort as possible into creating PRs. I'd like to convince
+you that even a small investment in improving them pays over-sized dividends to the project and your
+team members.
+
+# Background
+
+I'll start by admitting that because the PR is frequently at the very end of a development effort
+that we should pay respect to the mental exhaustion present. A project spike will start with a
+mountain of known unknowns, and some unknown unknowns as well, all of which an IC will be
+accountable for, in addition to writing code that is maintainable, tested, and covers edge cases. It
+can be exhausting and therefore understandable that we would feel the impulse to create the PR as
+quickly as possible to get this over the finish line. I hope you'll agree that by the end of this
+post that spending a bit more effort to create a better PR is worth it, and is probably not as much
+effort as one might think.
```

Comment:
You allude to it a bit later, but I feel the value of a well documented PR lives on past just the initial review too. If this is a service running in production, someone else on-call that didn't review the PR, may need to refer back to it when there's an outage and they are trying to understand what happened. If this is an OSS project, people may want to review the PRs that land in the next release to better understand how the project is changing. I guess to say, the time spent in making the PR clear to understand can pay out dividends many times over.